src/monitoring/mod.rs

```diff
@@ -117,6 +131,101 @@ pub fn increment_errors_emitted_counter() {
     prometheus::ERRORS_EMITTED_COUNTER.inc();
 }
 
+fn txid_tracking_db(chainstate_root_path: &str) -> Result<DBConn, DatabaseError> {
+    let mut path = PathBuf::from(chainstate_root_path);
+
+    path.push("tx_tracking.db");
+    let db_path = path.to_str().ok_or_else(|| DatabaseError::ParseError)?;
+
+    let mut create_flag = false;
+    let open_flags = if fs::metadata(&db_path).is_err() {
+        // need to create
+        create_flag = true;
+        OpenFlags::SQLITE_OPEN_READ_WRITE | OpenFlags::SQLITE_OPEN_CREATE
+    } else {
+        // can just open
+        OpenFlags::SQLITE_OPEN_READ_WRITE
+    };
+
+    let conn = DBConn::open_with_flags(&db_path, open_flags)?;
+
+    conn.busy_handler(Some(tx_busy_handler))?;
+
+    if create_flag {
+        conn.execute(
+            "CREATE TABLE processed_txids (txid TEXT NOT NULL PRIMARY KEY)",
+            rusqlite::NO_PARAMS,
+        )?;
+    }
+
+    Ok(conn)
+}
+
+fn txid_tracking_db_contains(conn: &DBConn, txid: &Txid) -> Result<bool, DatabaseError> {
+    let contains = conn
+        .query_row(
+            "SELECT 1 FROM processed_txids WHERE txid = ?",
+            &[txid],
+            |_row| Ok(true),
+        )
+        .optional()?
+        .is_some();
+    Ok(contains)
+}
+
+#[allow(unused_variables)]
+pub fn mempool_accepted(txid: &Txid, chainstate_root_path: &str) -> Result<(), DatabaseError> {
+    #[cfg(feature = "monitoring_prom")]
+    {
+        let tracking_db = txid_tracking_db(chainstate_root_path)?;
+
+        if txid_tracking_db_contains(&tracking_db, txid)? {
+            // processed by a previous block, do not track again
+            return Ok(());
+        }
+
+        prometheus::MEMPOOL_OUTSTANDING_TXS.inc();
+    }
+
+    Ok(())
+}
+
+#[allow(unused_variables)]
+pub fn log_transaction_processed(
```

Comment:
I think we'll need to think about what'll happen if the user:

* starts the node with the tx log turned on, 
* stops the node,
* restarts the node with the tx log turned off
* stops the node
* starts the node with the tx log turned on again.

Is handling missing transactions a concern?  If so, then is there a plan to either force the node to re-index all unprocessed transactions on subsequent boot-up (bitcoin does this IIRC), or should this method simply stop processing transactions altogether if the node ever starts up with the tx log turned off?

Reply:
I don't think there's necessarily an issue here -- tx's that are processed while tx log is turned off just won't appear in this table.

Reply:
Right; I'm asking if that's going to be a problem.

Reply:
I don't think so -- if someone is particularly interested in data for the tx log, they should run from a fresh start. If it turns out that the re-index support is a highly desired feature, we can always add support for that later on.

Reply:
Sounds good